src/server/db_setup.ts

```diff
@@ -182,3 +181,63 @@ export async function insertGameResults(
     // TODO: Could return latest DB results here
     return { status: 200, results: [] };
 }
+
+// Checks the existence of an ID in the database
+export async function checkGameIdExists(
+    gameId: string
+): Promise<boolean>{
+  // Check if the game already exists in the table
+  const res = await pool.query(
+      "SELECT * FROM game_results WHERE game_label = $1",
+      [gameId]
+  );
+
+
+  if(res.rows.length > 0) {
+      //Game ID already exists, return true
+      return true;
+  }
+
+  return false;
+
+
+}
+
+//to test data upload
+//when page is refreshed, submitted data shows up in raw results table
+// This is the new function that can handle multiple insertions
+export async function insertGameResults(
+    allReq: GameResultsForm[]
+): Promise<GameResultsFormResult> {
+    var result;
+    let allErrors: ErrorObject[] = [];
+
+    //Loops for additional game data
+    for (let req of allReq) {
+        const gameId = req.gameId.trim();
+
+        let gameIdExists = await checkGameIdExists(gameId);
```

Comment:
lets elevate this up / before the loop with insertions. If the last game is already in the DB, don't insert anything at all.